drivers/clock_control/clock_control_esp32_cam.c

```diff
@@ -0,0 +1,99 @@
+#define DT_DRV_COMPAT espressif_esp32_cam_clk
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(esp32_lcd_cam, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <soc/lcd_cam_struct.h>
+#include <soc/lcd_cam_reg.h>
+
+#define ESP32_CAM_PLL_F160M     160000000UL
+#define ESP32_CAM_PLL_F160M_SEL 3
+#define ESP32_CAM_CLK_OFF_SEL   0
+
+struct clock_control_esp32_cam_config {
+	const struct pinctrl_dev_config *pcfg;
+	const struct device *clk_dev;
+	struct device *clk_subsys;
+	uint32_t cam_clk;
+	uint8_t clock_sel;
+};
+
+static int enable_pheripheral_clock(const struct device *dev)
+{
+	const struct clock_control_esp32_cam_config *cfg = dev->config;
+	int ret = 0;
+
+	/* Enable peripheral */
+	if (!device_is_ready(cfg->clk_dev)) {
+		return -ENODEV;
+	}
+
+	return clock_control_on(cfg->clk_dev, cfg->clk_subsys);
+}
+
+static int set_camera_clock(uint32_t cam_clk)
+{
+	int ret = 0;
+
+	if (0 == cam_clk) {
+		LCD_CAM.cam_ctrl.cam_clk_sel =
```

Comment:
Can you use HAL API here as well?

Reply:
I left a comment in the [hal/cam PR. ](https://github.com/zephyrproject-rtos/hal_espressif/pull/316#discussion_r1718604142)